### DISCUSSION TITLE
Showing ownly readme file

### DISCUSSION THREAD

**Discussion by hainhtat** [community/community#23707] [Other Feature Feedback, Questions, & Ideas]
<p>I developed a react site with my custom domain and it’s only showing readme file.<br>
Here is my github repo : <a href="https://github.com/hainhtat/hainhtat.github.io" rel="noopener nofollow ugc">https://github.com/hainhtat/hainhtat.github.io</a><br>
Here is my domain : <a href="https://heinhtetaung.me" rel="noopener nofollow ugc">heinhtetaung.me</a></p>

**Comment by smithaborkar** [comment]
<p>Hello <a class="mention" href="/hainhtat">@hainhtat</a> ,</p>
<p>I checked your repo and looks like you have an html file at the root level which is good. <a href="https://github.com/hainhtat/hainhtat.github.io/blob/c70ae934e47b7a04defd08d0ad86092dda208d16/index.html" rel="noopener nofollow ugc">https://github.com/hainhtat/hainhtat.github.io/blob/c70ae934e47b7a04defd08d0ad86092dda208d16/index.html</a></p>
<p>If you want to bring up a <code>page</code> and make it serve from the root level, you would need to rename the <a href="https://github.com/hainhtat/hainhtat.github.io/blob/c70ae934e47b7a04defd08d0ad86092dda208d16/index.html" rel="noopener nofollow ugc">file</a> as <strong>index.html</strong> (lowercase). This should fix the issue.</p>
<p>Let us know if you still face any issues</p>

**Comment by hainhtat** [comment]
<p>Thank you for your help but I’m a bit confused. The <a href="https://github.com/hainhtat/hainhtat.github.io/blob/c70ae934e47b7a04defd08d0ad86092dda208d16/index.html" rel="noopener nofollow ugc">File</a> name is already index.html. I want to show that index.html from gh-pages branch but instead readme from main branch is shown.</p>

**Comment by otter-computer** [comment]
<p>It looks like your Pages site is set to build from the <code>master</code> branch. You’ll need to change this to the <code>gh-pages</code> branch from the repository settings before it will show the site hosted in that branch.</p>

**Comment by hainhtat** [comment]
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/3/f/3ffa48b8c5124e2db04028f71c80237a3c220b36.png" data-download-href="/uploads/short-url/97YjkuODoiWUXiI0zaTyr9EcKIm.png?dl=1" title="Screen Shot 2020-10-20 at 20.38.30" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/47028268/181078842-a017657b-7e62-4462-aede-4e6fb1144b45.png" alt="Screen Shot 2020-10-20 at 20.38.30" data-base62-sha1="97YjkuODoiWUXiI0zaTyr9EcKIm" width="690" height="431" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/3/f/3ffa48b8c5124e2db04028f71c80237a3c220b36_2_690x431.png, https://aws1.discourse-cdn.com/github/optimized/3X/3/f/3ffa48b8c5124e2db04028f71c80237a3c220b36_2_1035x646.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/3/f/3ffa48b8c5124e2db04028f71c80237a3c220b36_2_1380x862.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/3/f/3ffa48b8c5124e2db04028f71c80237a3c220b36_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screen Shot 2020-10-20 at 20.38.30</span><span class="informations">2880×1800 409 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div><br>
I have changed the master to gh-pages but it showed error 404 there isn’t  a github pages site here.

**Comment by otter-computer** [comment, marked as answer]
<p>You’ll need to link your custom domain again. GitHub attaches a custom domain to a repository by creating a file called <code>CNAME</code> in the root of your site’s publishing source that contains the domain. As you’ve changed the deployment branch that file is now not present, meaning your domain has become unlinked.</p>
<p>You can fix it by adding the custom domain in the repository settings!</p>

**Comment by hainhtat** [comment]
<p>I have added the CNAME file in the gh-pages and received a mail that says that my ip address is outdated. Here is my dns record</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/a/e/ae607c645f938efba1ff3d70ca1d9e70665bc810.png" data-download-href="/uploads/short-url/oSBNB2LHncMjImWOloKQkJUO2rK.png?dl=1" title="Screen Shot 2020-10-20 at 21.06.30" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/47028268/181078846-84646ed9-91be-4da6-95ee-58333b96acd9.png" alt="Screen Shot 2020-10-20 at 21.06.30" data-base62-sha1="oSBNB2LHncMjImWOloKQkJUO2rK" width="690" height="431" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/a/e/ae607c645f938efba1ff3d70ca1d9e70665bc810_2_690x431.png, https://aws1.discourse-cdn.com/github/optimized/3X/a/e/ae607c645f938efba1ff3d70ca1d9e70665bc810_2_1035x646.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/a/e/ae607c645f938efba1ff3d70ca1d9e70665bc810_2_1380x862.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/a/e/ae607c645f938efba1ff3d70ca1d9e70665bc810_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screen Shot 2020-10-20 at 21.06.30</span><span class="informations">2880×1800 355 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by hainhtat** [comment]
<p>Thanks for your help. I changed the dns settings and finally got it.</p>

**Comment by otter-computer** [comment]
<p>Nice! :smile: Yeah, I think those are the old Pages IPs, so you’d need to change your <code>A</code> records to the new IPs. For anyone else looking in future the new ones they’re listed in the instructions on this page here:</p>
<aside class="onebox allowlistedgeneric">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">
      <a href="https://docs.github.com/en/free-pro-team@latest/github/working-with-github-pages/managing-a-custom-domain-for-your-github-pages-site#configuring-an-apex-domain" target="_blank" rel="noopener">docs.github.com</a>
  </header>
  <article class="onebox-body">
    <img src="" class="thumbnail" width="16" height="16">

<h3><a href="https://docs.github.com/en/free-pro-team@latest/github/working-with-github-pages/managing-a-custom-domain-for-your-github-pages-site#configuring-an-apex-domain" target="_blank" rel="noopener">Managing a custom domain for your GitHub Pages site - GitHub Docs</a></h3>



  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>
